src/definitions/device_engagement/nfc_options.rs

```diff
@@ -26,46 +25,59 @@ impl TryFrom<CborValue> for NfcOptions {
     type Error = Error;
 
     fn try_from(v: CborValue) -> Result<Self, Error> {
-        let map: BTreeMap<CborValue, CborValue> = match v {
-            CborValue::Map(map) => Ok(map),
+        let map: BTreeMap<CborValue, CborValue> = match v.0 {
+            ciborium::Value::Map(map) => map
+                .into_iter()
+                .map(|(k, v)| {
+                    let k = CborValue::from(k)?;
+                    let v = CborValue::from(v)?;
```

Comment:
From what I can tell, this is the only place where you construct a `CborValue`, and I don't think it is necessary. You could use `ciborium::Value` directly.

Reply:
There are many places where we use `.try_into()`.

Reply:
Using primitive as key now.